### PR TITLE
Some improvemets in requirementslib for clarity

### PR DESCRIPTION
**Pull request by vshulcz**
### The issue
This PR addresses a couple of areas in the code where improvements in argument handling and path management were needed.
This PR does not fix a specific GitHub issue but addresses overall code quality.

### The fix

Removed an extra space after -i, added some mini docstrings, improved the path handling logic.

### The checklist

* [x] Associated issue: No specific issue, general improvement.
* [ ] A news fragment in the `news/` directory to describe this fix with the extension `.bugfix.rst`, `.feature.rst`, `.behavior.rst`, `.doc.rst`. `.vendor.rst`. or `.trivial.rst` (this will appear in the release changelog). Use semantic line breaks and name the file after the issue number or the PR #.

### CODE REVIEW

File: pipenv/utils/requirementslib.py [hunk 1]
```diff
@@ -654,7 +655,7 @@ def unpack_url(
         would ordinarily raise HashUnsupported) are allowed.
     """
     # non-editable vcs urls
-    if link.scheme in [
+    if link.scheme in {
```
Comment (matteius) [inline pipenv/utils/requirementslib.py:658]: While you are at it, can we update to use `VCS_SCHEMES` from constants?

Comment (vshulcz) [reply pipenv/utils/requirementslib.py]: Of course! I've now updated the code. I thought the differences with constant might cause problems

**Comment by oz123** [comment]
@vshulcz can you please add a news file?

**Comment by matteius** [comment]
I'd be fine merging this as-is considering the scope of the other changes and the comprehensive news fragment I added there.

**Pull request merged**